scripts/cmake/sdk_android.cmake

```diff
@@ -0,0 +1,134 @@
+    endif()
+else()
+    message(DEBUG "sdk_android: DEFOLD_SDK_ROOT not set; skipping packaged NDK detection")
+endif()
+
+# Helper: add toolchains from an Android SDK path
+function(_defold_android_sdk_add_toolchains SDK_PATH)
+    if(NOT EXISTS "${SDK_PATH}")
+        return()
+    endif()
+    file(GLOB _ndk_roots
+        "${SDK_PATH}/ndk/*"
+        "${SDK_PATH}/ndk-bundle"
+    )
+    foreach(_ndk IN LISTS _ndk_roots)
+        if(EXISTS "${_ndk}/build/cmake/android.toolchain.cmake")
+            list(APPEND _ANDROID_TOOLCHAINS "${_ndk}/build/cmake/android.toolchain.cmake")
+        elseif(EXISTS "${_ndk}/android.toolchain.cmake")
+            list(APPEND _ANDROID_TOOLCHAINS "${_ndk}/android.toolchain.cmake")
+        endif()
+    endforeach()
+    set(_ANDROID_TOOLCHAINS "${_ANDROID_TOOLCHAINS}" PARENT_SCOPE)
+endfunction()
+
+# 2) Local Android SDK via env vars
+if(NOT _ANDROID_TOOLCHAINS)
+    if(DEFINED ENV{ANDROID_SDK_ROOT})
+        _defold_android_sdk_add_toolchains("$ENV{ANDROID_SDK_ROOT}")
+    elseif(DEFINED ENV{ANDROID_HOME})
+        _defold_android_sdk_add_toolchains("$ENV{ANDROID_HOME}")
+    endif()
+endif()
+
+# 3) Well-known default SDK roots
+if(NOT _ANDROID_TOOLCHAINS)
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
+        _defold_android_sdk_add_toolchains("$ENV{HOME}/Library/Android/sdk")
+    elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+        _defold_android_sdk_add_toolchains("$ENV{HOME}/Android/Sdk")
+    elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+        if(DEFINED ENV{LOCALAPPDATA})
+            _defold_android_sdk_add_toolchains("$ENV{LOCALAPPDATA}/Android/Sdk")
+        endif()
+        if(DEFINED ENV{APPDATA})
+            _defold_android_sdk_add_toolchains("$ENV{APPDATA}/Android/Sdk")
+        endif()
+    endif()
+endif()
+
+# Pick the most recent toolchain file
+if(_ANDROID_TOOLCHAINS)
+    list(SORT _ANDROID_TOOLCHAINS COMPARE NATURAL)
+    list(REVERSE _ANDROID_TOOLCHAINS)
+    list(GET _ANDROID_TOOLCHAINS 0 _ANDROID_TOOLCHAIN_FILE)
+endif()
+
+if(EXISTS "${_ANDROID_TOOLCHAIN_FILE}")
+    get_filename_component(_toolchain_dir "${_ANDROID_TOOLCHAIN_FILE}" DIRECTORY)
+    # Typical path: <ndk_root>/build/cmake/android.toolchain.cmake
+    # NDK root is either parent (android.toolchain.cmake in root) or grandparent (build/cmake/...)
+    get_filename_component(_ndk_parent "${_toolchain_dir}" DIRECTORY)
+    if(EXISTS "${_ndk_parent}/source.properties")
+        set(_ANDROID_NDK_ROOT "${_ndk_parent}")
+    else()
+        get_filename_component(_ndk_grandparent "${_ndk_parent}" DIRECTORY)
+        set(_ANDROID_NDK_ROOT "${_ndk_grandparent}")
+    endif()
+
+    if(_SDKS_DIR AND _ANDROID_TOOLCHAIN_FILE MATCHES "^${_SDKS_DIR}.*")
+    defold_log("sdk_android: Using packaged Android NDK: ${_ANDROID_NDK_ROOT}")
+    else()
+    defold_log("sdk_android: Using local Android SDK NDK: ${_ANDROID_NDK_ROOT}")
+    endif()
+    defold_log("sdk_android: Toolchain file: ${_ANDROID_TOOLCHAIN_FILE}")
```

Comment:
As long as we choose correct API level (below), I don't think it should matter if the local install gets a newer build system (than r25b that we prepackage). E.g. I'd like to be able to use the locally installed Android versions on CI as well. Having prepackaged versions is a nuisance so if we can avoid it...